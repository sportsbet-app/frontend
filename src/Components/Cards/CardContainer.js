
import React from 'react'
import { Link, Route } from "react-router-dom";

//import components being implemented 
// import { UGameCard } from './UpcomingGameCard' //<-- currently not in use
import NFL from "../Sports/NFL"
import MLB from "../Sports/MLB";
import NCAA from "../Sports/NCAA";
import CFL from "../Sports/CFL";
import NBA from "../Sports/NBA";

//import styling
import textureBG from '../../Assets/texture.jpg'
import styled from 'styled-components'
import { colors, buttonSize } from '../../Theme/Variables'


//overall container
const CardHolder = styled.div`
  max-width: 100%;
  background: linear-gradient(to top, ${colors.darkGrey}, ${colors.darkGrey});
  color: #fff;
`

const Card = styled.div`
  width: 100%;
`

//styling for the page heading/title
const ContainerTitle = styled.div`
<<<<<<< HEAD
  background-color: ${colors.primary};
  color: #fff;
  width: 60%;
  max-height: 38px;
  display: flex;
  align-items: center;
  margin-bottom: 40px;

  h2 {
    font-size: 1.4rem;
    padding-left: 1.2rem;
    text-transform: uppercase;
=======
background-color: ${colors.primary};
color: #fff;
width: 60%;
max-height: 38px;
display: flex;
align-items: center;
margin-bottom: 40px;
    h2 {
        font-size: 1.4rem;
        padding-left: 1.2rem;
        text-transform: uppercase;
>>>>>>> 71b572c01d5099fbd9a1e6efa21f92b7e599b777
    }
`

const ContainerContent = styled.div`
  margin: 0 auto;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  padding: 2rem 0;
`

const CardContainer = props => {
  return (
    <CardHolder>
      <Card>
        <ContainerTitle>
          <h2>Quick Bets - NFL</h2>
        </ContainerTitle>
        <ContainerContent> 
          {/* maybe change all of these to links and add the routes in the router location(app.js) */}
          <Route path="/nfl" component={NFL} />
          <Route path="/mlb" component={MLB} />
          <Route path="/ncaa" component={NCAA} />
          <Route path="/cfl" component={CFL} />
          <Route path="/nba" component={NBA} />
        </ContainerContent>
      </Card>
    </CardHolder>
  )
}

export default CardContainer