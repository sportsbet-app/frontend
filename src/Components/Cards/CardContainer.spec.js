import React from 'react'
import { render } from '@testing-library/react'
//import renderer for snapshot
import renderer from 'react-test-renderer'

import CardContainer from './CardContainer'

//test to check that the app will display when called.
describe('<CardContainer />', () => {
  it('should display', () => {
    //render the component
    render(<CardContainer/>)
    
  })
  //create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<CardContainer />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
})