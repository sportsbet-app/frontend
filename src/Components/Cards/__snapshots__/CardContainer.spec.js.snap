// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContainer /> matches snapshot 1`] = `
<div
  className="sc-jzJRlG bgPJAT"
>
  <div
    className="sc-cSHVUG feWfX"
  >
    <div
      className="sc-kAzzGY ithIdY"
    >
      <h2>
        Quick Bets - NFL
      </h2>
    </div>
    <div
      className="sc-chPdSV kWeWpb"
    />
  </div>
</div>
`;
